fargate/check.py

```diff
@@ -0,0 +1,158 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import time
+
+# 3rd party
+import requests
+
+# project
+from checks import AgentCheck
+
+# Fargate related constants
+EVENT_TYPE = SOURCE_TYPE_NAME = 'fargate'
+API_ENDPOINT = 'http://169.254.170.2/v2'
+METADATA_ROUTE = '/metadata'
+STATS_ROUTE = '/stats'
+DEFAULT_TIMEOUT = 5
+
+# Default value is maxed out for some cgroup metrics
+CGROUP_NO_VALUE = 0x7ffffffffffff000
+
+# Do not collect these labels are we already have the info as tags
+LABEL_BLACKLIST = ["com.amazonaws.ecs.cluster", "com.amazonaws.ecs.container-name", "com.amazonaws.ecs.task-arn",
+                   "com.amazonaws.ecs.task-definition-family", "com.amazonaws.ecs.task-definition-version"]
+
+# Metrics constants
+MEMORY_GAUGE_METRICS = ['cache','mapped_file','rss','hierarchical_memory_limit','active_anon',
+                        'active_file','inactive_file','hierarchical_memsw_limit']
+MEMORY_RATE_METRICS = ['pgpgin', 'pgpgout', 'pgmajfault','pgfault']
+IO_METRICS = {'io_service_bytes_recursive':'fargate.io.bytes.', 'io_serviced_recursive':'fargate.io.ops.'}
+
+class FargateCheck(AgentCheck):
+
+    def check(self, instance):
+        timeout = float(instance.get('timeout', DEFAULT_TIMEOUT))
+        metadata_endpoint = API_ENDPOINT + METADATA_ROUTE
+        stats_endpoint = API_ENDPOINT + STATS_ROUTE
+
+        try:
+            request = requests.get(metadata_endpoint, timeout=timeout)
+        except requests.exceptions.Timeout as e:
+            msg = 'Fargate {} endpoint timed out after {} seconds'.format(metadata_endpoint, timeout)
+            self.service_check('fargate_check', AgentCheck.CRITICAL, message=msg)
+            self.log.error(msg)
+            return
+        except Exception as e:
```

Comment:
I'm not a huge fan of swallowing exceptions. Can we at least log it? Same in all the other `except` block, except when it's obvious like the timeout one above.